src/internal/metrics/metrics.go

```diff
@@ -238,11 +242,33 @@ func inputMetrics(input *pps.Input, metrics *Metrics) {
 	}
 }
 
+func (r *Reporter) attemptGetToken() bool {
+	if r.token != "" {
+		return true
+	}
+	// retry asynchronously here?
+	if resp, _ := r.env.AuthServer().GetRobotToken(context.Background(),
```

Comment:
I like the idea of generating a token for background jobs, rather than trying to create unauthenticated interfaces across services. Instead of a robot user for this, my recommendation would be to create a new internal API for creating tokens with a unique namespace, like `internal:`, so they're easily identifiable compared to user-created robots.

Reply:
Both of those sound like great ideas, thanks!